packages/ses-ava/src/ses-ava-test.js

```diff
@@ -23,21 +14,48 @@ const isPromise = maybePromise =>
   Promise.resolve(maybePromise) === maybePromise;
 
 /**
- * @type {LogCallError}
+ * @typedef {(...args: unknown[]) => void} Logger
+ */
+
+/**
+ * Calls `func(...args)` passing back approximately its outcome, but first
+ * logging any erroneous outcome to the `logger`, which defaults to
+ * `console.log`.
+ *
+ *    * If `func(...args)` returns a non-promise, silently return it.
+ *    * If `func(...args)` throws, log what was thrown and then rethrow it.
+ *    * If `func(...args)` returns a promise, immediately return a new
+ *      unresolved promise.
+ *       * If the first promise fulfills, silently fulfill the returned promise
+ *         even if the fulfillment was an error.
+ *       * If the first promise rejects, log the rejection reason and then
+ *         reject the returned promise with the same reason.
+ *
+ * The delayed rejection of the returned promise is an observable difference
+ * from directly calling `func(...args)` but will be equivalent enough for most
+ * purposes.
+ *
+ * TODO This function is useful independent of Ava, so consider moving it
+ * somewhere and exporting it for general reuse.
+ *
+ * @param {(...unknown) => unknown} func
+ * @param {unknown[]} args
+ * @param {string} name
+ * @param {Logger=} logError
  */
-const logErrorFirst = (func, args, name, logger = defaultLogger) => {
+const logErrorFirst = (func, args, name, logError = console.error) => {
```

Comment:
```suggestion
const logErrorFirst = (func, args, name, logError = (...args) => console.error(...args)) => {
```
just in case on a platform in which `console.error` is this-sensitive.

Reply:
The only platform Ava runs on is Node. They don't even intend to support Deno and if they did it woudl be using the Node-compatibility API https://github.com/avajs/ava/discussions/2743